modulemd/modulemd-yaml-util.c

```diff
@@ -592,15 +592,32 @@ modulemd_yaml_parse_string_set (yaml_parser_t *parser, GError **error)
         case YAML_SCALAR_EVENT:
           g_debug ("Parsing scalar: %s",
                    (const gchar *)event.data.scalar.value);
-          g_hash_table_add (result,
-                            g_strdup ((const gchar *)event.data.scalar.value));
 
-          if (!in_list)
+          if (in_list)
             {
-              /* We got a scalar instead of a sequence. Treat it as a list with
-               * a single entry
-               */
-              done = TRUE;
+              g_hash_table_add (
+                result, g_strdup ((const gchar *)event.data.scalar.value));
+            }
+          else
+            {
+              /* We got a scalar instead of a sequence. */
+              if (event.data.scalar.value && event.data.scalar.value[0])
+                {
+                  /* Treat it as a list with a single entry
+                   * if it's nonempty. */
+                  g_hash_table_add (
+                    result, g_strdup ((const gchar *)event.data.scalar.value));
+                  done = TRUE;
+                }
+              else
+                {
+                  /* Otherwise it's a bug in libyaml which misreports an empty
```

Comment:
It's not technically a bug in libyaml. The [YAML specification](https://yaml.org/spec/1.2.2/#72-empty-nodes) actually requires this explicitly. Best not to confuse a future reader with the assertion that it's a bug. No need to send for re-review just for this.

Reply:
I corrected the comment. Though I still understand the specification as describing two distinct cases, one resulting into an empty string (7.2) and the other one resulting to an undefined value (7.3). Interesting is that YAML::XS can tell the distinction despite being built on top of the very same libyaml.